tests/ui/diagnostic_namespace/on_const/misplaced_attr.stderr

```diff
@@ -15,18 +18,27 @@ error: `#[diagnostic::on_const]` can only be applied to trait impls
    |
 LL | #[diagnostic::on_const(message = "tadaa", note = "boing")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+LL |
+LL | impl const PartialEq for Foo {
+   | ---------------------------- not a trait impl
```

Comment:
We'll need to fix this to specify on `const` `impl`s that the annotation only works on non-`const` ones, but that can be done in a follow up PR.